graphql/src/main/java/io/jmix/graphql/datafetcher/MessagesDataFetcher.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.graphql.datafetcher;
+
+import graphql.schema.DataFetcher;
+import io.jmix.core.*;
+import io.jmix.core.accesscontext.CrudEntityContext;
+import io.jmix.core.accesscontext.EntityAttributeContext;
+import io.jmix.core.metamodel.model.MetaClass;
+import io.jmix.core.metamodel.model.MetaProperty;
+import io.jmix.graphql.schema.messages.MessageDetail;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component(value = MessagesDataFetcher.NAME)
+public class MessagesDataFetcher {
+    public static final String NAME = "gql_MessagesDataFetcher";
+
+    @Autowired
+    private Metadata metadata;
+    @Autowired
+    private MetadataTools metadataTools;
+    @Autowired
+    private AccessManager accessManager;
+    @Autowired
+    private MessageTools messageTools;
+    @Autowired
+    private Messages messages;
+
+    public DataFetcher<?> loadEntityMessages() {
```

Comment:
Looks like such code could be rewritten via stream
```suggestion
       public DataFetcher<?> loadEntityMessages() {
        return environment -> metadata.getSession().getClasses().stream()
                .flatMap(metaClass -> this.getEntityMessages(metaClass).stream())
                .collect(Collectors.toList());
    }
```